app/models/test_track/session.rb

```diff
@@ -52,41 +43,49 @@ def state_hash
   end
 
   def log_in!(identity, forget_current_visitor: nil)
-    identifier_type = identity.test_track_identifier_type
-    identifier_value = identity.test_track_identifier_value
-
-    @visitor = TestTrack::Visitor.new if forget_current_visitor
-    visitor.link_identifier!(identifier_type, identifier_value)
+    @unauthenticated_visitor = TestTrack::Visitor.new if forget_current_visitor
+    unauthenticated_visitor.link_identity!(identity)
+    visitors_by_identity[identity] = unauthenticated_visitor
```

Comment:
I changed the contract of `visitor.link_identifier!` to `visitor.link_identity!`. Identities work nice. Let's use them.